File: android/src/main/java/com/getcapacitor/community/classes/events/ReceivedEvent.java
package com.getcapacitor.community.classes.events;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import java.util.UUID;

public class ReceivedEvent extends MessageIDEvent {

    @NonNull
    private final String message;

    @Nullable
    private final String peerID;

    @Nullable
    private final Boolean isPrivate;

    @Nullable
    private final Boolean isRelay;

    public ReceivedEvent(
        @Nullable UUID messageID,
        @NonNull String message,
        @Nullable String peerID,
        @Nullable Boolean isPrivate,
        @Nullable Boolean isRelay
    ) {
        super(messageID);
        this.message = message;
        this.peerID = peerID;

        this.isPrivate = isPrivate;
        this.isRelay = isRelay;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = super.toJSObject();

        result.put("message", message);

        if (peerID != null) {
            result.put("peerID", peerID);
        }

        if (isPrivate != null) {
            result.put("isPrivate", isPrivate);
        }

        if (isRelay != null) {
            result.put("isRelay", isRelay);
        }

        return result;
    }
}

File: ios/Sources/BitchatPlugin/Classes/Events/ReceivedEvent.swift
import Foundation
import Capacitor

@objc public class ReceivedEvent: MessageIDEvent {
    let message: String

    let peerID: String?

    let isPrivate: Bool?
    let isRelay: Bool?

    init(_ messageID: UUID?, _ message: String, _ peerID: String?, _ isPrivate: Bool?, _ isRelay: Bool?) {
        self.message = message

        self.peerID = peerID

        self.isPrivate = isPrivate
        self.isRelay = isRelay

        super.init(messageID)
    }

    override public func toJSObject() -> JSObject {
        var result = super.toJSObject()

        result["message"] = message

        if let peerID = peerID {
            result["peerID"] = peerID
        }

        if let isPrivate = isPrivate {
            result["isPrivate"] = isPrivate
        }

        if let isRelay = isRelay {
            result["isRelay"] = isRelay
        }

        return result
    }
}
